src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1983,38 +1983,49 @@ public void RemoveProjectOutputPath_NoLock(SolutionChangeAccumulator solutionCha
 
         private void RefreshMetadataReferencesForFile(object sender, string fullFilePath)
         {
-            using (_gate.DisposableWait())
+            var asyncToken = _workspaceListener.BeginAsyncOperation(nameof(RefreshMetadataReferencesForFile));
```

Comment:
💡  Place this token inside the callback with `using var asyncToken`. This is acceptable because `RunWithShutdownBlockAsync` behaves like `JoinableTaskFactory.RunAsync`, where the callback is synchronously invoked on the current thread before returning. Only `Task.Run` requires the pattern currently used here.

Reply:
I admit the behavior of RunWithShutdownBlockAsync wasn't intuitive there; fixed this and another place we were doing the same thing.